test/integration/parachain/staging/sequential/vaults.test.ts

```diff
@@ -114,123 +111,202 @@ describe("vaultsAPI", () => {
+            let issuableAmount = await vault.getIssuableTokens();
+            issuableAmount = issuableAmount.mul(issuableAmountModifier);
+            await issueSingle(interBtcAPI, bitcoinCoreClient, oracleAccount, issuableAmount, vault_3_id);
+    
+            const currentVaultCollateralization =
+                await interBtcAPI.vaults.getVaultCollateralization(newAccountId(api, vault_3.address), collateralCurrencyIdLiteral);
+            if (currentVaultCollateralization === undefined) {
+                throw new Error("Collateralization is undefined");
+            }
+    
+            const collateralCurrencyTyped = collateralCurrency as Currency<CollateralUnit>;
+    
+            // The factor to adjust the exchange rate by. Calculated such that the resulting collateralization
+            // will be 90% of the premium redeem threshold. (e.g. 1.35 * 90% = 1.215)
+            const premiumRedeemThreshold = await interBtcAPI.vaults.getPremiumRedeemThreshold(collateralCurrency);
+            const modifyExchangeRateBy = premiumRedeemThreshold.mul(0.9).div(currentVaultCollateralization);
+    
+            const initialExchangeRate = await interBtcAPI.oracle.getExchangeRate(collateralCurrencyTyped);
+            // crash the exchange rate so that the vault falls below the premium redeem threshold
+            const exchangeRateValue = initialExchangeRate.toBig().div(modifyExchangeRateBy);
+            const mockExchangeRate = new ExchangeRate<
+                Bitcoin,
+                BitcoinUnit,
+                typeof collateralCurrencyTyped,
+                typeof collateralCurrencyTyped.units
+            >(Bitcoin, collateralCurrencyTyped, exchangeRateValue);
+    
+            // stub the oracle API to always return the new exchange rate
+            const stub = sinon.stub(interBtcAPI.oracle, "getExchangeRate")
+                .withArgs(sinon.match.any)
+                .returns(Promise.resolve(mockExchangeRate as any)); // "as any" to help eslint play nicely
+    
+            const premiumRedeemVaults = await interBtcAPI.vaults.getPremiumRedeemVaults();
+    
+            // Check that the stub has indeed been called at least once 
+            // If not, code has changed and our assumptions when mocking the oracle API are no longer valid
+            sinon.assert.called(stub);
+            sinon.restore();
+    
+            // real assertions here
+            assert.isAtLeast(premiumRedeemVaults.size, 1);
+
+            // locate the amount for the current vault
+            let premiumRedeemAmount: MonetaryAmount<WrappedCurrency, BitcoinUnit> | undefined = undefined;
+            for (const [vaultId, amount] of premiumRedeemVaults) {
+                if (encodeVaultId(vaultId) === encodeVaultId(vault_3_id)) {
+                    premiumRedeemAmount = amount;
+                    break;
+                }
+            }
```

Comment:
There is probably a more functional way to write this logic but that's probably more a nit from my side

Reply:
Right. A version of `array.find().map()` would do the trick.
Keeping this for my clean up task.